src/object.rs

```diff
@@ -133,6 +133,70 @@ impl JSObject {
             ctx: self.value.ctx,
         }
     }
+
+    /// Call this object considering it is a valid function.
+    ///
+    /// ```rust
+    /// # use javascriptcore::{JSContext, JSValue};
+    /// let ctx = JSContext::default();
+    /// let global = ctx.global_object().unwrap();
+    /// let math = global.get_property("Math").as_object().unwrap();
+    /// let pow = math.get_property("pow").as_object().unwrap();
+    ///
+    /// let result = pow.call_as_function(
+    ///     &ctx,
+    ///     None,
+    ///     &[JSValue::new_number(&ctx, 2.), JSValue::new_number(&ctx, 3.)],
+    /// ).unwrap();
+    ///
+    /// assert_eq!(result.as_number().unwrap(), 8.);
+    /// ```
+    pub fn call_as_function(
+        &self,
+        context: &JSContext,
+        this: Option<&JSObject>,
+        arguments: &[JSValue],
+    ) -> Result<JSValue, JSException> {
+        let arguments = arguments
+            .iter()
+            .map(|argument| argument.raw)
+            .collect::<Vec<_>>();
+        let mut exception: sys::JSValueRef = ptr::null_mut();
+
+        let result = unsafe {
+            sys::JSObjectCallAsFunction(
+                context.raw,
+                self.raw,
+                this.map(|this| this.raw).unwrap_or_else(ptr::null_mut),
+                arguments.len(),
+                arguments.as_slice().as_ptr(),
+                &mut exception,
+            )
+        };
+
+        if !exception.is_null() {
+            return Err(JSException {
+                value: JSValue {
+                    raw: exception,
+                    ctx: context.raw,
+                },
+            });
+        }
+
+        if result.is_null() {
+            return Err(JSException {
+                value: JSValue::new_string(
+                    &context,
```

Comment:
This should just be `context` to avoid the clippy warning since it is already a reference…